### DISCUSSION TITLE
Where is Incremental Static Regeneration saving the updates?

### DISCUSSION THREAD

**Discussion by colinsteidtmann** [vercel/next.js#35842] [Help]
I'm confused where ISR revalidate or unstable_revalidate() saves the updated file. 

I thought it saved it to the user's cache, then if the user cleared their cache then they would see the old static page generated during the last production build. However, when I cleared my cache to test this, I saw the recently updated page and not the one that I originally built for production. So, now my guess is that Next.js updates the production build (does it update some files under the cache folder in .next/cache?). If Next.Js updates the production build when revalidate is called, then what I originally pushed to github would be slightly different (stale) compared to the updated code running on a separate server (firebase for example), is that correct?

**Comment by icyJoseph** [comment]
Hi,

I think you are a bit confused. ISR basically takes note of the page definition, and re-runs it, server side, as needed. 

That's the same for simple revalidate, the `getStaticPaths` + `getStaticProps` combo, or manual `unstable_revalidate`. 

Why would there be any difference, between what you push to GitHub or the production server? Are you committing the outputs of running `next build`? ISR doesn't change the source code of the application.

The entire point of ISR is that the most recent build of any page, contains the latest, most up-to-date data, but that's up to the 3rd party providing such data. So that, if we run your GitHub code, after you've deployed, we should see, an even more up-to-date version of you data.

As to where it is saved, I do believe it is somewhere in `.next`, there's also a flag to control the size of the `ISR` cache, `isrMemoryCacheSize?: number`.

**Reply by colinsteidtmann** [reply]
Thanks. I think part of my confusion comes from not know how Next.js cache (or caches in general) actually work. Do you know of any resources or terms I can google to find out more?
